adding review process document (#33)
Kitware/SMQTK-Core: docs/review_process.rst

SMQTK Review Process
********************

The purpose of this document is to define the process for reviewing and integrating branches into SMQTK. This encompasses all SMQTK repositories.

See ``CONTRIBUTING.md`` for guidelines on contributing to SMQTK.

> See `release_process`_ for guidelines on the release process for SMQTK.

Author: No I was referring to the release process for new releases while this exclusively covers PR's

> See `release_process`_ for guidelines on the release process for SMQTK.

Reviewer: Oops! Apparently I dislexia'd "release" for "review".

.. _`release_process`: release_process.html

> .. contents:: The review process consisists of the following steps:

Reviewer: ```suggestion
.. contents:: The review process consists of the following steps:
```

> .. contents:: The review process consisists of the following steps:

Author: I think contents makes sense for larger files. Looking at this now, I think i want to adjust the depth so it looks more concise.

* ``LGTM Analysis`` runs a more advanced code analysis tool over the branch that can address issues that the submitter might not have noticed. Should LGTM find an issue, it will write a comment on the PR. The comment should be addressed by the submitter before continueing to submit for review.

Human Review
============

Once the automatic checks are either resolved or addressed, the submitted PR will need to go through a human review. Reviewers should add comments to provide feedback and raise potential issues. Should the PR pass their review, the reviewer should then indicate that it has their approval.

A review can still be requested before the checks are resolved, but the PR must be marked as a ``Draft``. Once the PR is in a mergable state, it will need to undergo a final review to ensure that there are no outstanding issues.

CI/MyPy/Pytest
==============

> * ``lint`` runs flake8 to quality check the code. You can run this check manually in your local repository with ``poetry run flake8``.

Reviewer: ```suggestion
* ``lint`` runs flake8 to quality check the code style. You can run this check manually in your local repository with ``poetry run flake8``.
```

> * ``MyPy`` performs a type check. You can run this check manually in your local respository with ``poetry run mypy``.

Reviewer: ```suggestion
* ``MyPy`` performs static type analysis. You can run this check manually in your local repository with ``poetry run mypy``.
```
